components/solr-access-service/api/src/main/java/org/phenotips/solr/HPOScriptService.java

```diff
@@ -78,11 +79,14 @@
             crt = nodes.poll();
             results.add(String.valueOf(crt.get(ID_FIELD_NAME)));
             @SuppressWarnings("unchecked")
-            List<String> parents = (List<String>) crt.get("is_a");
+            Object parents = crt.get("is_a");
             if (parents == null) {
                 continue;
             }
-            for (String pid : parents) {
+            if (parents instanceof String) {
+                parents = Collections.singletonList(parents);
+            }
+            for (String pid : (List<String>) parents) {
```

Comment:
Define two variables instead, `parents` stays of type `List<String>`, and `rawParents` as an object. Assign `parents` to either casted `rawParents`, or `Collections.singletonList(String.valueOf(rawParents))`.